docs/gaps/teams-proposal/README.md

```diff
@@ -0,0 +1,188 @@
+identify groups within a delegation (since they are implicit), as well as to
+modify all associated delegations when someone needs to be added to or removed
+from a group. With teams in gittuf, the delegating entity only needs to run an
+update to their team definition once, which will affect all delegations where
+the team is authorized.
+
+This GAP introduces the idea of a team in gittuf, in which multiple identities
+can be grouped together as part of a single team, and a single identity can
+be associated with multiple teams. 
+
+In addition to the definition of teams, the idea of wearing different "hats"
+must be supported. In some cases, people may be assigned to multiple teams, and
+must be able to select which team to represent when signing off a change using
+gittuf. For instance, a repository contributer Alice is a in 2 different teams:
+'security' and 'dev', where these two teams have different privileges and
+thresholds. In this scenario, Alice would wear her 'dev' hat during normal
+updates, and wear her 'security' hat only for emergency updates that require
+immediate action. As seen here, hats are a way for a person to select which team
+they wish to represent when signing off a change.
+
+## Specification
+A team type contains the following fields:
+* ID - A unique identifier of the team entity
+* Principals - A set of principals associated with this team
+* Threshold - A number of approvals required by principals associated to this
+  team
+
+The schema of a team would look like the following.
+```
+"<id>": {
+    "teamID": "<id>"
+    "principals": {
+        "<personID-1>": {
+            "personID": "<personID-1>",
+            "publicKeys": {
+                "<keyID1>": {
+                    "keyid": "<keyID1>",
+                    "keyid_hash_algorithms": null,
+                    "keytype": "sigstore-oidc",
+                    "keyval": {
+                        "identity": "<username>",
+                        "issuer": "<issuer>"
+                    },
+                    "scheme": "fulcio"
+                },
+                "<keyid2>>": ...
+            },
+            "associatedIdentities": {
+                "<integrationID1>": "<usernameA>",
+                "<integrationID2>": "<usernameB>",
+                ...
+            }
+            "customMetadata": {...}
+        },
+        "<personID-2>": {
+            ...
+        },
+        ...
+    }
+    "threshold": "<int>"
+}
+```
+
+A hat attestation is used when a user wishes to approve a change by attesting on
+behalf of a team. A hat authorization contains the following fields:
+* TargetRef - The name of the Git reference that will be updated
+* FromID - The ID of the Git reference before the change is applied.
+* TargetID - The ID of the Git reference after the change is applied.
+* PrincipalID - The attesting user's principal ID.
+* Hat - The attesting user's team they are attesting on behalf of.
```

Comment:
So, they do this after the fact?  Or would it make sense to have the hat as an optional field of their existing way to generate an attestation?  

What if someone makes an attestation without a hat?  What happens?  Is this all teams, no teams, a specific team, etc.?